frontend/dashboard/src/app/page.tsx

```diff
@@ -1,112 +1,153 @@
-import Image from "next/image";
+"use client";
+import {Button} from "@/components/ui/button";
+import {InputWithLabel} from "@/components/ui/inputWithLabel";
+import { SelectScrollable } from "@/components/ui/selectScrollable";
+import { College, Interest } from "@/const/dummyConst";
+import MultipleSelector from '@/components/ui/multiSelect';
+import { useForm, Controller } from "react-hook-form";
+import { z } from "zod";
+import {Option} from "@/types/option";
+import { useState } from 'react';
+
+type FormData = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  college: string;
+  interests: Option[];
+};
+
+const formSchema = z.object({
+  firstName: z.string().min(2),
+  lastName: z.string().min(2),
+  email: z.string().email({
+    message: "Please enter your email"
+  }),
+  college: z.string().min(1),
+  interests: z.array(z.object({
+    label: z.string(),
+    value: z.string()
+  })).nonempty(),
+})
 
 export default function Home() {
+  const [formDataJson, setFormDataJson] = useState<string>('');
+  const {
+    control,
+    handleSubmit,
+    formState: { errors }
+} = useForm<FormData>();
```

Comment:
Set the mode to be on onTouched so that the user receives live feedback

https://react-hook-form.com/docs/useform#mode

Reply:
React hook form recommends using "default values"

https://react-hook-form.com/docs/useform#defaultValues